pkg/odo/cli/deploy/deploy.go

```diff
@@ -129,9 +131,24 @@ func (o *DeployOptions) initDevfile() error {
 // Run contains the logic for the odo command
 func (o *DeployOptions) Run() error {
 	devfileObj := o.EnvSpecificInfo.GetDevfileObj()
+	devfileName := devfileObj.GetMetadataName()
 	path := filepath.Dir(o.EnvSpecificInfo.GetDevfilePath())
 	appName := o.GetApplication()
-	return o.clientset.DeployClient.Deploy(devfileObj, path, appName)
+	namespace := o.GetProject()
+
+	// Output what the command is doing / information
+	log.Title("Deploying the Devfile: "+devfileName,
+		"Namespace: "+namespace,
+		"CLI Version: "+version.VERSION)
```

Comment:
```suggestion
		"odo version: "+version.VERSION)
```

Reply:
It might be better to create a wrapper function with just `devfileName` and `namespace` parameters and have rest of the strings inside the function to ensure that the `odo dev` and `odo deploy` output is always i sync